tests/Serval.SpecFlowTests/StepDefinitions/MachineApiStepDefinitions.cs

```diff
@@ -0,0 +1,337 @@
+        {
+            if (engine.Name == user)
+                return engine.Id;
+        }
+        throw new ArgumentException($"No engine for user {user} available.");
+    }
+
+    public async Task<string> PostCorpusToEngine(
+        string engineId,
+        FileFormat fileFormat,
+        string[] filesToAdd,
+        string sourceLanguage,
+        string targetLanguage,
+        bool pretranslate
+    )
+    {
+        var sourceFiles = await PostFiles(filesToAdd, fileFormat, sourceLanguage);
+        var sourceFileConfig = new List<TranslationCorpusFileConfig>();
+
+        foreach (var item in sourceFiles.Select((file, i) => new { i, file }))
+        {
+            sourceFileConfig.Add(
+                new TranslationCorpusFileConfig { FileId = item.file.Id, TextId = filesToAdd[item.i] }
+            );
+        }
+
+        var targetFileConfig = new List<TranslationCorpusFileConfig>();
+        var targetFiles = await PostFiles(filesToAdd, fileFormat, targetLanguage);
+        foreach (var item in targetFiles.Select((file, i) => new { i, file }))
+        {
+            targetFileConfig.Add(
+                new TranslationCorpusFileConfig { FileId = item.file.Id, TextId = filesToAdd[item.i] }
+            );
+        }
+
+        var response = await translationEnginesClient.AddCorpusAsync(
+            id: engineId,
+            new TranslationCorpusConfig
+            {
+                Name = "None",
+                SourceFiles = sourceFileConfig,
+                SourceLanguage = sourceLanguage,
+                TargetFiles = targetFileConfig,
+                TargetLanguage = targetLanguage
+            }
+        );
+
+        if (pretranslate)
+        {
+            translationBuildConfig.Pretranslate!.Add(
+                new PretranslateCorpusConfig { CorpusId = response.Id, TextIds = filesToAdd.ToList() }
+            );
+        }
+
+        return response.Id;
+    }
+
+    private void ClearPretranslationConfig()
+    {
+        translationBuildConfig.Pretranslate = new List<PretranslateCorpusConfig>();
+    }
+
+    public async Task<List<DataFile>> PostFiles(IEnumerable<string> filesToAdd, FileFormat fileFormat, string language)
+    {
+        string languageFolder = Path.GetFullPath(
+            Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "../../..", "data", language)
+        );
+        if (!Directory.Exists(languageFolder))
+            throw new ArgumentException($"The langauge data directory {languageFolder} does not exist!");
+        // Collect files for the corpus
+        var files = Directory.GetFiles(languageFolder);
+        if (files.Length == 0)
+            throw new ArgumentException($"The langauge data directory {languageFolder} contains no files!");
+        var fileList = new List<DataFile>();
+        var all_files = await dataFilesClient.GetAllAsync();
+        var filename_to_id = (from file in all_files where file.Name is not null select file).ToLookup(
```

Comment:
Although this is a perfectly valid LINQ query, I stick to only functional LINQ, so that the code is consistent.

Reply:
I need to review this to familiarize myself with the difference.

Reply:
Updated.